### ISSUE TITLE
MaxWebsocketMessageSize is hard-coded

### ISSUE THREAD

**Issue by hoytech** [mattgodbolt/seasocks#93]
If you want bigger inbound messages you need to bump this up in `src/main/c/Connection.cpp` (and maybe also `ReadWriteBufferSize` -- dunno I did both), and then recompile library.

**Comment by mattgodbolt** [comment]
Good point! (wow, how big are your messages!) We should make this configurable though :)

**Comment by hoytech** [comment]
Hehe usually a lot smaller but in one case about 500 bytes bigger than that limit, which is how I noticed this :)

Outbound messages are sometimes 1M pre-compression and no problems there, this only seems to affect in-bound.

**Comment by A-J-Bauer** [comment]
I also noticed this when trying to send a large text file.
For testing I changed ReadWriteBufferSize, MaxWebsocketMessageSize in Connection.cpp to 256 * 1024.

This works for Firefox. Chrome however seems to send chunks for large strings. Presumably setting FIN bit 1 for the first, FIN bit 0 for following and FIN bit 1 for the last one. This however is not supported in HybiPacketDecoder.cpp (Line 46).

I could try to fix this but after thinking about it I figure that since the memory on my little Raspberry is limited it might be better to send the file line by line and also write it to the sd card (not frequently) instead of keeping it in memory. This might be a thing to think about in general I guess since Seasocks most likely is used on embedded devices with limited resources.

**Comment by stazio** [comment]
What is the state of this? Seasocks hangs for that one connection for me if I try to send too big of a string.

**Comment by A-J-Bauer** [comment]
You can check out my Pull Request "Support for fragmented Messages" (which I couldn't get thru test insanity) and see if it works for you.
